### DISCUSSION TITLE
Is it possible to train two different NERs sourced from en_core_web_md ?

### DISCUSSION THREAD

**Discussion by oliviercwa** [explosion/spaCy#9231] [Help: Coding & Implementations] [answered]
Something like:

```
pipeline = ["tok2vec", "ner1", "ner2]

[components.ner1]
source = "en_core_web_md"

[components.ner2]
source = "en_core_web_md"
...
```

This does not work as is and I could not find a way to indicate that the "source name" in "en_core_web_md" is "ner"

Ideally we could do something like

```
pipeline = ["tok2vec", "ner1", "ner2]

[components.ner1]
source_model = "en_core_web_md"
source_name = "ner"

[components.ner2]
source_model = "en_core_web_md"
source_name = "ner"
...
```

Is there a clean way to achieve this ?

**Comment by adrianeboyd** [comment, marked as answer]
It's:

```ini
[components.ner2]
source = "en_core_web_md"
component = "ner"
```

Edit to fix both config settings, not just the `component`.

**Comment by oliviercwa** [comment]
Unfortunately this causes an exception when training.

I have a single NER components defined 
```
[components.ner2]
source = "en_core_web_md"
component = "ner"
```

Causes the following exception

```
  File "spacy\spacy\training\loop.py", line 122, in train
    raise e
  File "spacy\spacy\training\loop.py", line 112, in train
    log_step(info if is_best_checkpoint is not None else None)
  File "spacy\spacy\training\loggers.py", line 63, in log_step
    for pipe_name in logged_pipes
  File "spacy\spacy\training\loggers.py", line 63, in <listcomp>
    for pipe_name in logged_pipes
KeyError: 'ner2'
```

**Reply by adrianeboyd** [reply]
That does look like a bug, we will take a look.

However, it doesn't really make sense to train two `ner` components in the same pipeline at the same time, since you can only have one source of NER annotations in your training corpus.

There's a slight possibility that the second `ner` component might make a few additional annotations because it will be starting from a different `Doc` state when the pipeline is run (existing entity spans in the `Doc`), but this would be an odd/inefficient setup overall.
